### DISCUSSION TITLE
Input Mask Ability?

### DISCUSSION THREAD

**Discussion by X3Technology** [DNNCommunity/Dnn.StructuredContent#75] [Ideas]
How useful would it be to be able to set an input mask on textbox type inputs?  In the setup options we could provide a list of common mask (phone number, zip code, SSN, etc.) and also have the ability for the user to provide a custom mask and a custom validation message.

Useful?

**Comment by mitchelsellers** [comment]
This is one of the most common data-input pattern issues, very helpful.

**Comment by valadas** [comment]
Yeah, but those common masks would vary a lot by culture. I think some sort of regex would be easier

**Reply by X3Technology** [reply]
Regex would certainly be an option.  But I suspect a lot of users don't know how to use Regex.  I could also see a more generic template mask as well.  

I know formats of the common things like phone numbers, etc. will vary by country, but I still feel like it makes sense to a least offer them as defaults.  The user can always choose the "custom mask" option if none of the pre-selects work.

**Reply by mitchelsellers** [reply]
I think even basic masking syntax would be good. This is how many controls do it.  

For example 

000-000-0000  would be a us phone number, or you could use the # sign?

or 

00/00/0000 for a date

**Comment by WillStrohl** [comment]
There must be existing patterns or plugins that already take care of the localization part of the mask.

**Comment by X3Technology** [comment]
OK.  I think the basic requirement is to idea to have a way for the user to provide the mask.  

I was thinking it would be a nice ease of use feature to provide a "quick list" of predefined masks like phone number, SSN , zip code, etc., but I see the localization issues.  That a nice to have, not a must have.  Perhaps a potential solution here is to provide a way for the admin to change the options in the quick list.  We could have some defaults, but they could change them for whatever locale they are in.

I will work on implementing the basic mask functionality first.

**Comment by X3Technology** [comment]
This is done
